### PR TITLE
Style/private payment redesign

### PR DESCRIPTION
**Pull request by Boyuan-Deng**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [ ] Test code changes manually, and describe your procedure in a comment on this PR
- [ ] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/components/Balance/index.tsx [hunk 1]
```diff
@@ -16,8 +16,8 @@ const Balance: React.FC<IBalanceProps> = ({
 }) => {
   return (
     <div id="balanceText" className={className}>
-      Balance:&nbsp;<strong>{balance}</strong>
-      {loader ? <DotLoader /> : null}
+      Balance:&nbsp;{balance}
```
Comment (Kevingislason) [inline src/components/Balance/index.tsx:19]: a bit more elegant to define this string outside of the JSX

```js
`Balance: ${balance}`
```

File: src/pages/SendPage/SendContext/index.tsx [hunk 2]
```diff
@@ -29,6 +29,7 @@ export const SendContextProvider = (props) => {
     changeExternalAccount
   } = useExternalAccount();
   const privateWallet = usePrivateWallet();
+  const [ senderInputValue, setSenderInputValue] = useState('');
```
Comment (Kevingislason) [inline src/pages/SendPage/SendContext/index.tsx:32]: Input value state should be local to the input component, this is the convention throughout manta-front-end. Other components shouldn't have to care about the string typed into the sender input, only the sanitized / validated `Balance` senderTargetBalance

**Review by Kevingislason — COMMENTED** [review]
Nice work. I think most of the remaining work on this involves having some conversations with Chloe. A few things:

-The height of the central component changes very slightly when we toggle private / public, which would be best to avoid.
-I think it's a bit awkward that the address input form (private transfer and public transfeer) is shorter than the balance display (public to private and private to public). There is a lot of dead space beneath the address input. I think it might be more elegant to center the address input vertically between the transact button and the public private toggle. Could also make that component taller, but I'm not sure that would look good. Might want to ask Chloe what she thinks.
-If I hover over the bottom component on public to private or private to public, my cursor changes such that it looks like I can input text, but actually the input is not interactable. I think  it would be helpful to avoid this confusion.
-When we click on the up and down arrow thing, it should toggle private and public i.e. to public becomes to private and vice versa
-I think we need to figure out exactly what colors / gradients that the mockup is using (the most recent one on sketch). Also need to clarify whether the colors on the mockup are preferred to the colors on the current UI, or whether they were just used as an approximation.
-Same thing with the font, not sure if the font in the mock is the font our designers intend for us to use, or if the font we are currently using is just not available.
-The design displays validations on the transact button rather than in red text beneath the forms as previously, I think this feature still needs to be implemented

File: src/pages/SendPage/SendButton.tsx [hunk 3]
```diff
@@ -14,15 +15,92 @@ const SendButton = () => {
...
+    } else if (userHasSufficientFunds() === false) {
+      // insufficient balance
+      return (
+        <div className="py-2 unselectable-text text-center text-white gradient-button filter brightness-50 rounded-lg w-full cursor-not-allowed">
+          Insuffient balance
+        </div>
+      );
+    } else if (
+      receiverAddress === null &&
+      (isPrivateTransfer || isPublicTransfer)
```
Comment (Kevingislason) [inline src/pages/SendPage/SendButton.tsx:59]: these are functions

File: src/pages/SendPage/SendButton.tsx [hunk 4]
```diff
@@ -14,15 +15,92 @@ const SendButton = () => {
...
+        isPrivateTransfer ? 'zkAddress' : 'address'
+      }`;
+      return (
+        <div className="py-2 unselectable-text text-center text-white gradient-button filter brightness-50 rounded-lg w-full cursor-not-allowed">
+          {message}
+        </div>
+      );
+    } else if (
+      receiverAddress === false &&
+      (isPrivateTransfer || isPublicTransfer)
```
Comment (Kevingislason) [inline src/pages/SendPage/SendButton.tsx:71]: these are functions

File: src/pages/SendPage/SendButton.tsx [hunk 5]
```diff
@@ -14,15 +15,92 @@ const SendButton = () => {
...
+        <div className="py-2 unselectable-text text-center text-white gradient-button filter brightness-50 rounded-lg w-full cursor-not-allowed">
+          Insuffient balance
+        </div>
+      );
+    } else if (
+      receiverAddress === null &&
+      (isPrivateTransfer || isPublicTransfer)
+    ) {
+      const message = `Enter Recipients ${
+        isPrivateTransfer ? 'zkAddress' : 'address'
```
Comment (Kevingislason) [inline src/pages/SendPage/SendButton.tsx:62]: isPrivateTransfer is a function

File: src/pages/SendPage/SendButton.tsx [hunk 6]
```diff
@@ -14,15 +15,92 @@ const SendButton = () => {
...
+      return (
+        <div className="py-2 unselectable-text text-center text-white gradient-button filter brightness-50 rounded-lg w-full cursor-not-allowed">
+          {message}
+        </div>
+      );
+    } else if (
+      receiverAddress === false &&
+      (isPrivateTransfer || isPublicTransfer)
+    ) {
+      const message = `Invalid ${isPrivateTransfer ? 'zkAddress' : 'address'}`;
```
Comment (Kevingislason) [inline src/pages/SendPage/SendButton.tsx:73]: isPrivateTransfer is a function

File: src/pages/SendPage/SendContext/index.tsx (hunk 2 above)
Comment (Kevingislason) [reply src/pages/SendPage/SendContext/index.tsx:32]: bump

File: src/pages/SendPage/SendFromForm/index.tsx [hunk 7]
```diff
@@ -9,34 +9,34 @@ const SendFromForm = () => {
...
+  // }
 
-  let warningText = null;
-  if (txWouldDepleteSuggestedMinFeeBalance()) {
-    warningText = `You need ${senderAssetType.ticker} to pay fees; consider retaining a small balance.`
-  }
+  // let warningText = null;
+  // if (txWouldDepleteSuggestedMinFeeBalance()) {
+  //   warningText = `You need ${senderAssetType.ticker} to pay fees; consider retaining a small balance.`
+  // }
```
Comment (Kevingislason) [inline src/pages/SendPage/SendFromForm/index.tsx:24]: You can delete these comments

We will want to think of how to handle `txWouldDepleteSuggestedMinFeeBalance` in a follow up, but need design feedback

File: src/pages/SendPage/SendFromForm/index.tsx [hunk 8]
```diff
@@ -9,34 +9,34 @@ const SendFromForm = () => {
...
+            prefix="sender"
+          />
+        </div>
       </div>
       <SendAssetSelect />
-      <FormErrorText
-        errorText={errorText}
-        warningText={warningText}
-      />
+      {/* <FormErrorText errorText={errorText} warningText={warningText} /> */}
```
Comment (Kevingislason) [inline src/pages/SendPage/SendFromForm/index.tsx:39]: Can delete now

**Review by Kevingislason — COMMENTED** [review]
Looks GREAT! ! Really professional looking implementation.

A few loose ends:
Screen Shot 2022-11-27 at 11 25 55 PM
Screen Shot 2022-11-27 at 11 38 43 PM
- [ ] Balance display in the components overflows if the balance is too high or during initial sync for private assets
- [ ] We still need to add validations for the public wallet and the signer being disconnected; see the figma design; note also that the transact button changes color to match the button corresponding to the missing account
- [ ] The public to public address validation shows "Enter Recipients zk address", but it should say "Enter Recipients Substrate Address" (if that fits)
- [ ] The public to private form shows "Enter Recipients zk address" if the signer isn't connected
- [ ] I don't think we want to get rid of the dot loaders (although I think they would look more in line with out design if they weren't bold). I think we want to display '--'  only if we aren't meant to display a loader, but we have no balance to display either.
- [ ] If you open the account selector, it looks a bit messed up on the Stake page
- [ ] Collators table on staking page is all white now
- [ ] The transaction success modal is slightly discolored in the center
- [ ] The recipient address input placeholders should say "Enter Recipient ZkAddress" / "Enter Recipient Substrate Address" instead of just "address" (if that all fits)
- [ ] The address select dropdown should be very slightly wider so that it doesn't scroll sideways.
- [ ] Select Max can be solid blue instead of gradient per the design

Nice to have
- [ ] Ideally, we show a loader on the processing component, as in the design. The loader would hang around until the transaction is finish.

**Review by AlexParshh — APPROVED** [review]
UI looks great

**Comment by Kevingislason** [comment]
Moved to https://github.com/Manta-Network/manta-front-end/pull/320

**Pull request closed without merging**